include/rocksdb/metadata.h

```diff
@@ -54,60 +132,42 @@ struct SstFileMetaData {
         being_compacted(_being_compacted),
         num_entries(0),
         num_deletions(0),
-        temperature(_temperature),
         oldest_blob_file_number(_oldest_blob_file_number),
         oldest_ancester_time(_oldest_ancester_time),
-        file_creation_time(_file_creation_time),
-        file_checksum(_file_checksum),
-        file_checksum_func_name(_file_checksum_func_name) {}
+        file_creation_time(_file_creation_time) {
+    file_number = _file_number;
+    file_type = kTableFile;
+    size = _size;
+    trim_to_size = false;
+    temperature = _temperature;
+    file_checksum = _file_checksum;
+    file_checksum_func_name = _file_checksum_func_name;
+  }
 
-  // File size in bytes.
-  uint64_t size;
-  // The name of the file.
-  std::string name;
-  // The id of the file.
-  uint64_t file_number;
-  // The full path where the file locates.
-  std::string db_path;
-
-  SequenceNumber smallest_seqno;  // Smallest sequence number in file.
-  SequenceNumber largest_seqno;   // Largest sequence number in file.
-  std::string smallestkey;        // Smallest user defined key in the file.
-  std::string largestkey;         // Largest user defined key in the file.
-  uint64_t num_reads_sampled;     // How many times the file is read.
-  bool being_compacted;  // true if the file is currently being compacted.
+  SequenceNumber smallest_seqno = 0;  // Smallest sequence number in file.
+  SequenceNumber largest_seqno = 0;   // Largest sequence number in file.
+  std::string smallestkey;            // Smallest user defined key in the file.
+  std::string largestkey;             // Largest user defined key in the file.
+  uint64_t num_reads_sampled = 0;     // How many times the file is read.
+  bool being_compacted =
+      false;  // true if the file is currently being compacted.
```

Comment:
Is there a reason to initialize these here?  Looks like they are already covered in the constructor.

Reply:
Yes, for the default constructor, because it's much more concise and maintainable than listing all the field names again.